sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/SqlOperators.java

```diff
@@ -171,6 +172,12 @@
           x -> createTypeFactory().createSqlType(SqlTypeName.BIGINT),
           new UdafImpl<>(new BeamBuiltinAggregations.BitXOr<Number>()));
 
+  public static final SqlOperator COUNTIF =
+      createUdafOperator(
+          "countif",
+          x -> createTypeFactory().createSqlType(SqlTypeName.BOOLEAN),
```

Comment:
Hi Sonam, I believe the type mismatch error comes from this line.

The type specified here should be the *return* type of `COUNT_IF`, which is `BIGINT` instead of `BOOLEAN`.